dotnet/src/Agents/OpenAI/OpenAIResponseAgent.cs

```diff
@@ -0,0 +1,137 @@
+    {
+        Verify.NotNull(messages);
+
+        var agentThread = await this.EnsureThreadExistsWithMessagesAsync(messages, thread, cancellationToken).ConfigureAwait(false);
+
+        // Invoke responses with the updated chat history.
+        var chatHistory = new ChatHistory();
+        chatHistory.AddRange(messages);
+        var invokeResults = ResponseThreadActions.InvokeAsync(
+            this,
+            chatHistory,
+            agentThread,
+            options ?? new OpenAIAssistantAgentInvokeOptions(),
+            cancellationToken);
+
+        // Notify the thread of new messages and return them to the caller.
+        await foreach (var result in invokeResults.ConfigureAwait(false))
+        {
+            await this.NotifyThreadOfNewMessage(agentThread, result, cancellationToken).ConfigureAwait(false);
+            yield return new(result, agentThread);
+        }
+    }
+
+    /// <inheritdoc/>
+    public async override IAsyncEnumerable<AgentResponseItem<StreamingChatMessageContent>> InvokeStreamingAsync(ICollection<ChatMessageContent> messages, AgentThread? thread = null, AgentInvokeOptions? options = null, [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        Verify.NotNull(messages);
+
+        var agentThread = await this.EnsureThreadExistsWithMessagesAsync(messages, thread, cancellationToken).ConfigureAwait(false);
+
+        // Invoke responses with the updated chat history.
+        var chatHistory = new ChatHistory();
+        chatHistory.AddRange(messages);
+        int messageCount = chatHistory.Count;
+        var invokeResults = ResponseThreadActions.InvokeStreamingAsync(
+            this,
+            chatHistory,
+            agentThread,
+            options,
+            cancellationToken);
+
+        // Return streaming chat message content to the caller.
+        await foreach (var result in invokeResults.ConfigureAwait(false))
+        {
+            yield return new(result, agentThread);
+        }
+
+        // Notify the thread of new messages
+        for (int i = messageCount; i < chatHistory.Count; i++)
+        {
+            await this.NotifyThreadOfNewMessage(agentThread, chatHistory[i], cancellationToken).ConfigureAwait(false);
+
+            if (options?.OnIntermediateMessage is not null)
+            {
+                await options.OnIntermediateMessage(chatHistory[i]).ConfigureAwait(false);
+            }
+        }
+    }
+
+    /// <inheritdoc/>
+    [Experimental("SKEXP0110")]
+    [ExcludeFromCodeCoverage]
+    protected override Task<AgentChannel> CreateChannelAsync(CancellationToken cancellationToken)
+    {
+        throw new NotImplementedException("API will be removed in a future release.");
+    }
+
+    /// <inheritdoc/>
+    [Experimental("SKEXP0110")]
```

Comment:
Also mark as obsolete?

```suggestion
    [Experimental("SKEXP0110")]
    [Obsolete("Use XYZ instead")]
```